### PR TITLE
Bump alloy to 0.8.0

### PR DESCRIPTION
**Pull request by moricho**
Bumped alloy related dependencies
- `alloy`: 0.8.0
- `core`: 0.8.15
- `op-alloy`: 0.8.0
- `revm-inspector`: 0.13.0

The main changes in this PR are to support `Typed2718` trait introduced in https://github.com/alloy-rs/alloy/pull/1746

### CODE REVIEW

**Comment by mattsse** [comment]
supersedes #13077

File: crates/primitives/src/block.rs [hunk 1]
```diff
@@ -612,19 +612,19 @@ impl<T: Transaction> BlockBody<T> {
     /// Returns whether or not the block body contains any blob transactions.
     #[inline]
     pub fn has_blob_transactions(&self) -> bool {
-        self.transactions.iter().any(|tx| tx.is_eip4844())
+        self.transactions.iter().any(|tx| Typed2718::is_eip4844(tx))
```
Comment (klkvr) [inline crates/primitives/src/block.rs:615]: let's just remove those methods from the reth `Transaction` trait https://github.com/paradigmxyz/reth/blob/fcff53a6f8aa22cba143ed2202e93be7129f9aee/crates/primitives-traits/src/transaction/mod.rs#L35-L63

then we can invoke this normally as just `tx.is_eip4844()`

File: crates/optimism/primitives/src/transaction/signed.rs [hunk 2]
```diff
@@ -228,7 +229,7 @@ impl alloy_rlp::Encodable for OpTransactionSigned {
 
     fn length(&self) -> usize {
         let mut payload_length = self.encode_2718_len();
-        if !self.is_legacy() {
+        if !Encodable2718::is_legacy(self) {
```
Comment (moricho) [inline crates/optimism/primitives/src/transaction/signed.rs:232]: In some parts, we have to specify `Encodable2718` or `Typed2718` specifically, but the result will be the same.
Should we consolidate the `is_legacy` method to one or the other?

File: crates/primitives/src/block.rs (hunk 1 above)
Comment (moricho) [reply crates/primitives/src/block.rs]: thanks, fixed it in https://github.com/paradigmxyz/reth/pull/13268/commits/fab7998728860c23f4852b29501944834a975d7d

File: crates/rpc/rpc-server-types/src/result.rs [hunk 3]
```diff
@@ -162,7 +162,6 @@ pub fn block_id_to_str(id: BlockId) -> String {
                 format!("hash {}", h.block_hash)
             }
         }
-        BlockId::Number(n) if n.is_number() => format!("number {n}"),
         BlockId::Number(n) => format!("{n}"),
```
Comment (moricho) [inline crates/rpc/rpc-server-types/src/result.rs:165]: Due to https://github.com/alloy-rs/alloy/pull/1765

File: crates/net/eth-wire-types/src/broadcast.rs [hunk 4]
```diff
@@ -432,13 +432,13 @@ impl Decodable for NewPooledTransactionHashes68 {
...
                 got: msg.types.len(),
-            })
+            });
         }
         if msg.hashes.len() != msg.sizes.len() {
             return Err(alloy_rlp::Error::ListLengthMismatch {
                 expected: msg.hashes.len(),
                 got: msg.sizes.len(),
-            })
+            });
```
Comment (klkvr) [inline crates/net/eth-wire-types/src/broadcast.rs:441]: please revert unrelated formatting changes

File: crates/optimism/primitives/src/transaction/signed.rs (hunk 2 above)
Comment (klkvr) [reply crates/optimism/primitives/src/transaction/signed.rs:232]: right, I think Encodable2718 should just have Typed2718 as supertrait

File: crates/primitives-traits/src/transaction/mod.rs [hunk 5]
```diff
@@ -49,18 +46,6 @@ pub trait Transaction:
     fn is_eip1559(&self) -> bool {
         self.ty() == EIP1559_TX_TYPE_ID
     }
-
-    /// Returns true if the transaction is an EIP-4844 transaction.
-    #[inline]
```
Comment (klkvr) [inline crates/primitives-traits/src/transaction/mod.rs:54]: the other methods above can be removed as well

**Review by klkvr — DISMISSED** [review]

File: crates/net/eth-wire-types/src/broadcast.rs (hunk 4 above)
Comment (moricho) [reply crates/net/eth-wire-types/src/broadcast.rs]: Got it, reverted the changes: https://github.com/paradigmxyz/reth/pull/13268/commits/2413cb2a437ba9c8af038564fb82bb0cc2588f86

**Review by mattsse — APPROVED** [review]
lgtm

@klkvr making Typed2718 a super trait of Encodeable is perhaps not worth it but would be nicer
